config/crd/bases/opentelemetry.io_targetallocators.yaml

```diff
@@ -3284,6 +3313,8 @@ spec:
                   type: object
                 type: array
                 x-kubernetes-list-type: atomic
+            required:
+            - managementState
```

Comment:
Does this mean that users will always need to provide the `.spec.managementState` field when creating an `OpenTelemetryCollector`?

Reply:
No, `managementState` has a default, and that applies if it's not set by the user. Otherwise, all the E2E tests would fail.

Worth noting that I had to change one test which did a `kubectl replace`. That command does not apply defaults, and in that case you do need to set `managementState` manually. I just changed it into an `apply` instead.